add support for fifo queues

chrisfarms opened the pull request:
updates the broker to correctly support and test fifo queues.

fifo queues have a strange requirement that the queue names must end with a ".fifo" suffix, since we did not explicitly test the fifo queue in the integration tests this went unnoticed.

the integration tests have been extended to now run two passes of the full lifecycle ... once for "standard" and once for "fifo" queues. Since fifo queues require a MessageGroupID/DeduplicationID set in the SendMessage calls we need to tweak the send payload per test as well as the provision params.

To support the weird "fifo" extension in names, queue names are now generated by functions and conform to the following pattern:

```
$PREFIX-$UUID-$KIND.$EXT
```

Where:

* `$PREFIX` is something like "paas-sqs-broker"
* `$UUID` is the service instance id
* `$KIND` is the primary/secondary marker
* `$EXT` is either `""` or `".fifo"` depending on the FIFO-ness

philandstuff commented on sqs/queue_template_test.go:
```diff
@@ -32,18 +32,41 @@ var _ = Describe("QueueTemplateBuilder", func() {
 			builder.QueueName = "q-name-a"
 		})
-		It("should set primary queue name", func() {
+		It("should set primary queue name with a .stnd extension", func() {
```
```suggestion
		It("should set primary queue name without a .fifo extension", func() {
```

philandstuff replied:
and similarly below

chrisfarms replied:
ah thanks ... that was an abandoned path

philandstuff reviewed (approved):
Looks good.

Can we add support for content-based deduplication (in this PR or otherwise)?

Is there support for [running the integration tests in parallel](https://onsi.github.io/ginkgo/#parallel-specs)?  Can we update the README to show how to do so?  Can we update the platform tests similarly?  This feels like a perfect use case for test case parallelism.

chrisfarms commented:
RE: ContentBasedDeduplication....

I'd rather not address it in this PR, since this was just about the fifo-ness.

However, from what I understand, when ContentDeduplication is enabled:

* You do not have to specify an message id, and one will be generated for you based on the message content
* If you _do_ specify a message id ... then it overrides the content-based one generated for you

When ContentDeduplication is disabled:

* You MUST specify a message id or it is an error


Soooo ... is this really a useful option?, we could just leave it turned on, since in that mode you get to pick based on if you set DeduplicationMessageId in the messsage. :thinking: thoughts?

philandstuff commented:
sure, if team paas is happy without, I'm happy without 👍

chrisfarms commented:
Added note about running tests in parallel by giving example of running from the ginkgo cli ... and changed the `make test` thing to use that method along with the `-steam` option, so concourse output doesn't just leave you waiting
